src/methods/checkTomlForSep.ts

```diff
@@ -0,0 +1,62 @@
+import { log } from "helpers/log";
+import { getHomeDomainFromAssetIssuer } from "methods/getHomeDomainFromAssetIssuer";
+import { getToml } from "methods/getToml";
+import { TomlFields, AnyObject } from "types/types.d";
+
+export const checkTomlForSep = async ({
+  sepName,
+  assetIssuer,
+  requiredKeys,
+  networkUrl,
+  homeDomain,
+}: {
+  sepName: string;
+  assetIssuer: string;
+  requiredKeys: TomlFields[];
+  networkUrl: string;
+  homeDomain?: string;
+}) => {
+  let homeDomainParam = homeDomain;
+
+  if (!homeDomainParam) {
+    homeDomainParam = await getHomeDomainFromAssetIssuer({
+      assetIssuer,
+      networkUrl,
+    });
+  }
+
+  if (!homeDomainParam) {
+    throw new Error("Home domain was not found and is required");
+  }
```

Comment:
I think a more appropriate pattern would be to raise an error from `getHomeDomainFromAssetIssuer()` instead of checking if `homeDomainParam` is defined after calling a function that looks like it should always return one.

This also matches the pattern used in `getToml()`